Src/ILGPU.Tests/ExchangeBufferOperations.cs

```diff
@@ -0,0 +1,204 @@
+﻿using ILGPU.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace ILGPU.Tests
+{
+    public abstract class ExchangeBufferOperations : TestBase
+    {
+
+        private const int Length = 32;
+
+        protected ExchangeBufferOperations(ITestOutputHelper output, TestContext testContext)
+            : base(output, testContext)
+        { }
+
+        internal static void CopyKernel(Index1 index, ArrayView<long> data)
+        {
+            data[index] -= 5;
+        }
+
+        [Theory]
+        [InlineData(10)]
+        [InlineData(int.MaxValue)]
+        [InlineData(int.MinValue)]
+        [KernelMethod(nameof(CopyKernel))]
+        public void Copy(long constant)
+        {
+            using var exchangeBuffer = Accelerator.AllocateExchangeBuffer<long>(Length);
+            for (int i = 0; i < Length; i++)
+            {
+                exchangeBuffer[new Index1(i)] = constant;
+            }
+            exchangeBuffer.CopyToAccelerator();
+
+            var expected = new long[Length];
+            for (int i = 0; i < Length; i++)
+            {
+                expected[i] = constant - 5;
+            }
+
+            Accelerator.Synchronize();
+            Execute(exchangeBuffer.Extent, exchangeBuffer.View);
+            exchangeBuffer.CopyFromAccelerator();
+            Accelerator.Synchronize();
+
+            var data = exchangeBuffer.CPUView;
+            Assert.Equal(data.Length, expected.Length);
+            for (int i = 0; i < Length; i++)
+            {
+                Assert.Equal(expected[i], data[i]);
+            }
+        }
+
+        internal static void Copy2DKernel(Index2 index, ArrayView<long, Index2> data)
+        {
+            data[index] -= 5;
+        }
+
+        [Theory]
+        [InlineData(10)]
+        [InlineData(int.MaxValue)]
+        [InlineData(int.MinValue)]
+        [KernelMethod(nameof(Copy2DKernel))]
+        public void Copy2D(long constant)
```

Comment:
Some here. I am wondering whether we can find an alternative way of testing the exchange buffers since their purpose is to transfer data and do not require specific kernel operations?

Reply:
Sure, I guess we could do no operation at all. As I mentioned above, the point of these tests was to assert that the CPUView would get updated. If you find this unnecessary I can build a much cleaner test just requiring copy.

Reply:
Yeah, I got your idea. I just wanted to point out that we try to minimize the required "compiler capabilities" per test. Anyway, I think this makes sense in your case - furthermore, we might want to polish the `[InlineData]` attributes.

Reply:
Anyways, for now I've decided to use MemberData to reduce duplicate code. If you mean something different ou can reopen the conversation I think.